internal/wasm/host.go

```diff
@@ -67,28 +67,29 @@ func (c *ModuleContext) Function(name string) publicwasm.Function {
 	if err != nil {
 		return nil
 	}
-	return &function{c: c, f: exp.Function}
+	return &exportedFunction{module: c, function: exp.Function}
 }
 
-// function wraps FunctionInstance to implement wasm.Function as otherwise there's a signature clash on Call.
-type function struct {
-	c *ModuleContext
-	f *FunctionInstance
+// exportedFunction wraps FunctionInstance so that it is called in context of the exporting module.
```

Comment:
clarified the naming here as the more important point is that this assigns the correct module context